index.ts

```diff
@@ -127,42 +114,77 @@ try {
       }
     }
 
-    const {stdout} = await getExecOutput('npx', [
-      'changeset',
-      'publish',
-      '--no-git-tags',
-      '--snapshot',
-      '--tag',
-      'snapshot',
-    ]);
-
-    const newTags = Array.from(stdout.matchAll(/New tag:\s+([^\s\n]+)/g)).map(
-      ([_, tag]) => tag,
-    );
+    await exec('npx', ['changeset', 'version', '--snapshot', 'snapshot']);
+
+    const {packages} = await getPackages(process.cwd());
+    const snapshots = [];
+    packages.forEach(({packageJson}) => {
+      const {name, version} = packageJson;
+      if (version.includes('snapshot')) snapshots.push(`${name}@${version}`);
+    });
 
-    if (!newTags.length) {
+    if (!snapshots.length) {
       throw new Error('Changeset publish did not create new tags.');
     }
 
-    const multiple = newTags.length > 1;
+    if (branch) {
+      await exec('git', ['add', '.']);
+      await exec('git', [
+        'commit',
+        '-m',
+        `${payload.comment.user.login}-${payload.issue.number}-snapshot`,
+      ]);
+      await exec('git', ['checkout', '-b', branch]);
+      await exec('git', ['push', '--force', 'origin', branch]);
+    } else {
+      if (!process.env.NPM_TOKEN) {
+        throw new Error(
+          'Please provide the NPM_TOKEN to the snapit GitHub action',
+        );
+      }
+
+      await exec(
+        'bash',
+        [
+          '-c',
+          `echo "//registry.npmjs.org/:_authToken=${process.env.NPM_TOKEN}" > "$HOME/.npmrc"`,
+        ],
+        silentOption,
+      );
+
+      await exec('npx', [
+        'changeset',
+        'publish',
+        '--no-git-tags',
+        '--snapshot',
+        '--tag',
+        'snapshot',
+      ]);
+    }
+
+    const multiple = snapshots.length > 1;
+
+    const introMessage = branch
+      ? `Your snapshot${multiple ? 's are' : 'is'} being published.**\n\n`
+      : `Your snapshot${multiple ? 's have' : 'has'} been published to npm.**\n\n`;
```

Comment:
We don't know when the branch deploys will be published so we change the phrasing for this approach.